### ISSUE TITLE
Autocomplete fails on inferred type

### ISSUE THREAD

**Issue by andrew-w-ross** [microsoft/TypeScript#26384]
**TypeScript Version:**  3.1.0-dev.20180810

**Search Terms:** Autocomplete inferred 

**Code**
```typescript
type ValueTypes = 1 | 2 | 3;

export type RecordValues<Props extends string | number | symbol> = Record<
  Props,
  ValueTypes
>;

export function identityFn<RV extends RecordValues<keyof RV>>(value: RV) {
  return value;
}

const recordValue: RecordValues<"asdf"> = {
  asdf: 2 //Autotcomplete works fine for the value here
};

identityFn({
  asdf: 1 //Autocomplete shows every type available
});
```
**Expected behavior:** 

For the autocomplete for the value on the second comment to be the same as the first.
Please note I'm not talking about the keys but the values.

**Actual behavior:**

Autocomplete for the value suggests every type available.

**Playground Link:**  [link](https://www.typescriptlang.org/play/#src=type%20ValueTypes%20%3D%201%20%7C%202%20%7C%203%3B%0D%0A%0D%0Aexport%20type%20RecordValues%3CProps%20extends%20string%20%7C%20number%20%7C%20symbol%3E%20%3D%20Record%3C%0D%0A%20%20Props%2C%0D%0A%20%20ValueTypes%0D%0A%3E%3B%0D%0A%0D%0Aexport%20function%20identityFn%3CRV%20extends%20RecordValues%3Ckeyof%20RV%3E%3E(value%3A%20RV)%20%7B%0D%0A%20%20return%20value%3B%0D%0A%7D%0D%0A%0D%0Aconst%20recordValue%3A%20RecordValues%3C%22asdf%22%3E%20%3D%20%7B%0D%0A%20%20asdf%3A%202%20%2F%2FAutotcompletion%20works%20fine%20here%0D%0A%7D%3B%0D%0A%0D%0AidentityFn(%7B%0D%0A%20%20asdf%3A%201%20%2F%2F%20Try%20autocomplete%20here%0D%0A%7D)%3B%0D%0A)
**Related Issues:** #14841

**Comment by sheetalkamat** [comment]
This doesn't repro with typescript@next

**Comment by andrew-w-ross** [comment]
@sheetalkamat I just tried with with 3.2.0-dev.20181110 this is still broken.

**Comment by andy-ms** [comment]
The issue is that in the first example, we show `1`, `2`, and `3` as recommended completions. For the second example we don't get a contextual type and don't have recommended completions.

**Comment by sheetalkamat** [comment]
@andrew-w-ross Are you still seeing this with typescript@next? 
![quickinfo](https://user-images.githubusercontent.com/8052792/50662100-1d074600-0f5a-11e9-9520-65cfe65a0016.gif)

If yes can you please provide tsserver log.
You can upload logs by following these steps:
1. With "typescript.tsserver.log": "verbose" enabled
2. Restart vscode or vscode insiders
3. Reproduce the problem
4. You can share the logs using
    - either by sharing logs in the folder that gets opened by running `open tsserver log` from command palette 
    or
    - Run ```code --upload-logs``` on the command line (or ```code-insiders --upload-logs``` if using vscode insiders) and follow instructions and Then share the log file ID

**Comment by andrew-w-ross** [comment]
@sheetalkamat Yeah the behaviour you're getting is perfect. Unfortunately I'm not getting that. Tried it on stable vs code and insiders with ts `3.3.0-dev.20190104` the original behaviour. 
Here is the log  id 3618733104.

**Comment by sheetalkamat** [comment]
@andrew-w-ross It seems the issue only repros if on line you haven't typed `1` yet (making it error type). Thanks a lot since now we have the repro.

**Comment by sheetalkamat** [comment]
@andrew-w-ross no. Sorry forgot to update the label.

**Comment by alexn-s** [comment]
is there any update on this issue?

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
